pkg/epp/controller/pod_reconciler.go

```diff
@@ -67,8 +67,12 @@ func (c *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 }
 
 func (c *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Filter specific inference pool
+	p := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		return object.GetNamespace() == c.Namespace
+	})
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&corev1.Pod{}).
+		For(&corev1.Pod{}).WithEventFilter(p).
```

Comment:
btw, an eventFilter is still a client side filter, a more scalable approach is to set the filter server side for the namespace, and we can do that via ByObject cache option [1], we can do that for all other objects we reconcile on as well, wdyt? 

[1] https://github.com/kubernetes-sigs/controller-runtime/blob/main/designs/cache_options.md

Reply:
Good suggestion